zvariant/src/signature.rs

```diff
@@ -433,25 +433,41 @@ impl<'de> Visitor<'de> for SignatureVisitor {
     }
 }
 
-fn ensure_correct_signature_str(signature: &[u8]) -> Result<()> {
-    if signature.len() > 255 {
-        return Err(serde::de::Error::invalid_length(
-            signature.len(),
-            &"<= 255 characters",
-        ));
-    }
+struct SignatureIter<'a> {
```

Comment:
why not just implement `Iterator` for `SigntureParser` itself?

Reply:
why not?